sympy/core/kind.py

```diff
@@ -178,3 +185,195 @@ def __repr__(self):
         return "BooleanKind"
 
 BooleanKind = BooleanKind()
+
+
+class KindDispatcher:
+    """
+    Dispatcher to select a kind from multiple kinds by binary dispatching.
+
+    .. notes::
+       This approach is experimental, and can be replaced or deleted in the future.
+
+    Explanation
+    ===========
+
+    SymPy object's kind vaguely represents the algebraic structure where the
```

Comment:
Is there something in the docs that can be linked to for `Kind`? It would be good to have a clickable link here in the docs.

Reply:
Add the link to docstring.